TutorLizard.BusinessLogic/Services/TutorService.cs

```diff
@@ -14,39 +14,30 @@ public TutorService(ITutorDataAccess dataAccess, IUserIdentityService userIdenti
     }
     public Ad CreateAd(string subject, string title, string description)
     {
-        // return created Ad
-        // return empty with id = 0 to indicate something went wrong
-        return new Ad()
+        int? tutorId = _userIdentityService.GetUserId();
+        if (tutorId is null)
         {
-            // this is only for tests
-            Id = 1,
-            Subject = subject,
-            Title = title,
-            Description = description
-        };
+            return new Ad()
+            { Id = 0 };
+        }
+
+        return _dataAccess.CreateAd((int)tutorId, subject, title, description);
+
     }
     public ScheduleItem CreateScheduleItem(int adId, DateTime dateTime)
     {
-        // return created ScheduleItem
-        // return empty with id = 0 to indicate something went wrong
-        return new ScheduleItem()
-        {
-            // this is only for tests
-            Id = 1,
-            AdId = adId,
-            DateTime = dateTime
-        };
+        return _dataAccess.CreateScheduleItem(adId, dateTime);
     }
     public Ad? GetAdById(int adId)
     {
         // return Ad (from _dataAccess) with provided adId
         // return null if no such Ad
-        return new Ad()
+        Ad? ad = _dataAccess.GetAdById(adId);
+        if (ad == null)
```

Comment:
This if statement changes nothing nothing. In situation that `ad` is null `return ad` will return null anyway.
However, it's a good place to do such a check if you had something else to take into account.
It's slightly better to do `is null` rather than `== null`, but until we start overriding `==` operator it makes no difference.

Reply:
It stays for now.